packages/core/src/connectors/mock.ts

```diff
@@ -178,6 +163,24 @@ export class MockConnector extends Connector {
         return [];
       case "wallet_requestAccounts":
         return [this._account.address];
+      case "wallet_addStarknetChain":
+        return true;
+      case "wallet_switchStarknetChain":
+        return true;
+      case "wallet_addDeclareTransaction":
+        // TODO
+        return {
+          class_hash: "",
+          transaction_hash: "",
+        } satisfies AddDeclareTransactionResult;
```

Comment:
I think we can delegate these calls to the account? `account.declare` returns a type that should have the correct fields.